ui/debug/src/js/grid.js

```diff
@@ -0,0 +1,120 @@
+/* eslint-env browser */
+
+const $newBlock = (() => {
+  return $('<div>', {
+    css: {
+      width: '40px',
+      height: '40px',
+      float: 'left',
+      'font-size': '35px',
+    },
+  });
+});
+
+class Grid {
+  constructor(game, $container) {
+    this.game = game;
+    this.width = game.width;
+    this.height = game.height;
+    this.$blocks = [];
+    this.$previewBlocks = [];
+    this.$chainNumber = $('<p>');
+    this.$time = $('<p>');
+    this.$score = $('<p>');
+
+    for (let y = 0; y < this.height; ++y) {
+      const $row = $('<div>', {css: {display: 'table'}});
+      $container.append($row);
+      for (let x = 0; x < this.width; ++x) {
+        const $block = $newBlock();
+        $row.append($block);
+        this.$blocks.push($block);
+      }
+    }
+    const $previewRow = $('<div>', {
+      css: {
+        display: 'table',
+        opacity: 0.5,
+        'margin-bottom': '10px',
+      }
+    });
+    $container.append($previewRow);
+    for (let x = 0; x < this.width; ++x) {
+      const $block = $newBlock();
+      $previewRow.append($block);
+      this.$previewBlocks.push($block);
+    }
+
+    this.$blocks.forEach(($block, index) => {
```

Comment:
Not sure whether it would be more efficient to just construct jQuery objects instead of ordinary arrays.